src/main/scala/com/gu/salesfoce/messageHandler/MessageHandlerLambda.scala

```diff
@@ -0,0 +1,111 @@
+package com.gu.salesfoce.messageHandler
+
+import java.io.{ByteArrayInputStream, InputStream, OutputStream}
+import java.util.concurrent.Executors
+import javax.xml.bind.JAXBContext
+import javax.xml.soap.MessageFactory
+
+import com.amazonaws.services.lambda.runtime.Context
+import com.amazonaws.services.sqs.model.SendMessageResult
+import com.gu.salesfoce.messageHandler.APIGatewayResponse._
+import com.gu.salesfoce.messageHandler.ResponseModels.{ApiResponse, Headers}
+import com.sforce.soap._2005._09.outbound._
+import play.api.libs.json.{JsValue, Json}
+
+import scala.collection.JavaConversions._
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.{Failure, Try}
+
+trait RealDependencies {
+  val queueClient = SqsClient
+}
+
+trait MessageHandler extends Logging {
+  def queueClient: QueueClient
+
+  val queueName = s"salesforce-outbound-messages-${Config.stage}"
+  val ThreadCount = 10
+  implicit val executionContext = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(ThreadCount))
+
+  val okXml =
+    """<?xml version="1.0" encoding="UTF-8"?>
+      |<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/">
+      |	<soapenv:Body>
+      |		<notificationsResponse xmlns="http://soap.sforce.com/2005/09/outbound">
+      |			<Ack>true</Ack>
+      |		</notificationsResponse>
+      |	</soapenv:Body>
+      |</soapenv:Envelope>
+    """.stripMargin
+
+  val okResponse = ApiResponse("200", Headers(), okXml)
+
+  def parseMessage(requestBody: String): List[ContactNotification] = {
+    val is = new ByteArrayInputStream(requestBody.getBytes)
+    val messageFactory = MessageFactory.newInstance()
+    val soapMessage = messageFactory.createMessage(null, is)
+    val body = soapMessage.getSOAPBody
+    val jc = JAXBContext.newInstance(classOf[Notifications])
+    val unmarshaller = jc.createUnmarshaller()
+    val je = unmarshaller.unmarshal(body.extractContentAsDocument(), classOf[Notifications])
+    val notifications = je.getValue()
+    notifications.getNotification.toList
+  }
+
+  case class QueueMessage(contactId: String)
+
+  implicit val messageFormat = Json.format[QueueMessage]
+
+  def credentialsAreValid(inputEvent: JsValue): Boolean = {
```

Comment:
When handling outbound messages from SF in members-data-api, we perform some kind of check against the Organisation Id (as well as using a 'secret' in the query string): https://github.com/guardian/members-data-api/blob/master/membership-attribute-service/app/parsers/Salesforce.scala#L42

I'm not sure if we want to consider adding this check here too?